tests/unit_tests/pyanaconda_tests/modules/subscription/test_subscription_tasks.py

```diff
@@ -287,6 +289,50 @@ def test_set_rhsm_config_tast_restore_default_value(self):
 
         mock_config_proxy.SetAll.assert_called_once_with(expected_dict, "")
 
+    def test_set_rhsm_config_task_not_satellite(self):
+        """Test the SetRHSMConfigurationTask task - not-satellite prefix handling."""
+        # if the subscription request has the no-satellite prefix, it should be stripped
+        # before the server hostname value is sent to RHSM
+        mock_config_proxy = Mock()
+        # RHSM config default values
+        default_config = {
+            SetRHSMConfigurationTask.CONFIG_KEY_SERVER_HOSTNAME: "server.example.com",
+            SetRHSMConfigurationTask.CONFIG_KEY_SERVER_PROXY_HOSTNAME: "proxy.example.com",
+            SetRHSMConfigurationTask.CONFIG_KEY_SERVER_PROXY_PORT: "1000",
+            SetRHSMConfigurationTask.CONFIG_KEY_SERVER_PROXY_USER: "foo_user",
+            SetRHSMConfigurationTask.CONFIG_KEY_SERVER_PROXY_PASSWORD: "foo_password",
+            SetRHSMConfigurationTask.CONFIG_KEY_RHSM_BASEURL: "cdn.example.com",
+            "key_anaconda_does_not_use_1": "foo1",
+            "key_anaconda_does_not_use_2": "foo2"
+        }
+        # a representative subscription request
+        request = SubscriptionRequest()
+        request.type = SUBSCRIPTION_REQUEST_TYPE_ORG_KEY
+        request.organization = "123456789"
+        request.account_username = "foo_user"
+        request.server_hostname = SERVER_HOSTNAME_NOT_SATELLITE_PREFIX + "candlepin.foo.com"
```

Comment:
I wonder if using constants in the tests is correct way. 

I mean this way you won't be able to catch a typo in the constant value. I would recommend to avoid this.

Reply:
Like, we can change that.